### ISSUE TITLE
3.3.5 procCharges self-sub bug

### ISSUE THREAD

**Issue by winways** [TrinityCore/TrinityCore#21903]
Description: Aura::PrepareProcToTrigger has a wrong logic judgment:
`
if 
(
    IsUsingCharges() && (
    !eventInfo.GetSpellInfo() || 
    !eventInfo.GetSpellInfo()->HasAttribute(SPELL_ATTR6_DONT_CONSUME_PROC_CHARGES))
)
{
    --m_procCharges;
    SetNeedClientUpdateForTargets();
}
`

Current behaviour: this judgment doesn't jump into "--m_procCharges"

Expected behaviour: this judgment jumps into "--m_procCharges"

Steps to reproduce the problem: 
1.Compile a new core with RelWithDebInfo configuration.
2.Up your database to date.
3.Modify spell 46916 duration time index to 21 and then cast 47475 under aura 46916.
4.you can see 46916 will not disappear.

Branch(es): 3.3.5

TC rev. hash/commit: 9fb11fc8f2c770e1d6261baf050a2f6c0ce690ee

Operating system: windows 7

**Comment by Shauren** [comment]
Not a bug. 47475 has SPELL_ATTR6_DONT_CONSUME_PROC_CHARGES

**Comment by winways** [comment]
47475 will drop one charge of 46916 

it should be this
`    if (IsUsingCharges() && (this->m_spellInfo == nullptr || !(this->m_spellInfo->HasAttribute(SPELL_ATTR6_DONT_CONSUME_PROC_CHARGES))))

`

**Comment by joschiwald** [comment]
Should be fixed with #21260.

As workaround, try this:
```sql
UPDATE `spell_proc` SET `AttributeMask`=0x2 WHERE `SpellId`=46916;
```
